src/librustc_codegen_ssa/back/link.rs

```diff
@@ -899,7 +906,47 @@ pub fn print_native_static_libs(sess: &Session, all_native_libs: &[NativeLibrary
     }
 }
 
+#[cfg(all(windows, target_env = "gnu"))]
+fn get_compiler_libs_path(sess: &Session) -> Option<PathBuf> {
+    let system_libs = SYSTEM_LIBS.lock().unwrap().clone();
+    if let Some(compiler_libs_path) = system_libs {
+        println!("cache: hit");
+        return Some(compiler_libs_path);
+    } else {
+        println!("cache: miss");
+        let compiler = if let Some(linker) = &sess.opts.cg.linker {
+            linker.clone().into_os_string()
+        } else if let Some(linker) = &sess.target.target.options.linker {
+            linker.into()
+        } else {
+            return None;
+        };
+        if let Ok(output) = Command::new(compiler)
+        .arg("-print-file-name=crt2.o")
+        .output()
+        {
+            if let Some(compiler_libs_path) =
+                PathBuf::from(std::str::from_utf8(&output.stdout).unwrap()).parent()
+            {
+                let compiler_libs_path = fix_windows_verbatim_for_gcc(compiler_libs_path);
+                *SYSTEM_LIBS.lock().unwrap() = Some(compiler_libs_path.clone());
+                return Some(compiler_libs_path);
+            }
+        }
+    }
+    None
+}
+
 pub fn get_file_path(sess: &Session, name: &str) -> PathBuf {
+    #[cfg(all(windows, target_env = "gnu"))]
```

Comment:
This applies to when the *host* is `pc-windows-gnu` and not when the *target* is `pc-windows-gnu`.

Reply:
That's the problem I mentioned in the PR.
It's also required when cross compiling but it doesn't work yet.